src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelDiscoveryEventsHandler.cs

```diff
@@ -151,5 +163,21 @@ private void ConvertToRawMessageAndSend(string messageType, object payload)
             var rawMessage = this.dataSerializer.SerializePayload(messageType, payload);
             this.actualDiscoveryEventsHandler.HandleRawMessage(rawMessage);
         }
+
+        /// <summary>
+        /// Filters dictionary based on condition if source was fully discovered or not
+        /// </summary>
+        /// <param name="discoveryStatus">discoveryStatus indicates if source was fully or partially discovered</param>
+        /// <returns></returns>
+        private IReadOnlyCollection<string> getFilteredSources(DiscoveryStatus discoveryStatus)
```

Comment:
Nit pick: this should be named according to Pascal Case, i.e. `GetFilteredSources`.